opencog/atoms/base/ClassServer.h

```diff
@@ -210,6 +253,21 @@ class ClassServer
 
 ClassServer& classserver();
 
+#define DEFINE_LINK_FACTORY(CNAME,CTYPE)                          \
+                                                                  \
```

Comment:
CocneptNode doesn't need this. Most atoms don't -- only the atoms that corresponds to C++ classes, and need to have thier constructors run, when the atom is created.   Right now, these are some of the link types, and NumberNode and TypeNode.   The Number& TypeNdes are still done by hand (they don't interact badly with the python bindings)